pkg/azure/access/helpers/disk.go

```diff
@@ -32,7 +31,7 @@ const (
 // DeleteDisk deletes disk for passed in resourceGroup and diskName.
 // NOTE: All calls to this Azure API are instrumented as prometheus metric.
 func DeleteDisk(ctx context.Context, client *armcompute.DisksClient, resourceGroup, diskName string) (err error) {
-	defer instrument.RecordAzAPIMetric(err, diskDeleteServiceLabel, time.Now())
+	defer instrument.AZAPIMetricRecorderFn(diskDeleteServiceLabel, &err)()
```

Comment:
I think you need to pass the time here and not initialize it in the function.

Reply:
Not required as discussed, function value and parameters are evaluated eagerly.
https://go.dev/ref/spec#Defer_statements